src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1608,4 +1612,186 @@ public void fetchMetricsBasedOnDatasource(KruizeObject kruizeObject, Timestamp i
             throw new Exception(AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.METRIC_EXCEPTION + e.getMessage());
         }
     }
+
+    /**
+     * Fetches metrics based on the specified datasource using queries from the metricProfile for the given time interval.
+     *
+     * @param kruizeObject
+     * @param interval_end_time
+     * @param interval_start_time
+     * @param dataSourceInfo
+     * @throws Exception
+     */
+    public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, Timestamp interval_end_time, Timestamp interval_start_time, DataSourceInfo dataSourceInfo) throws Exception {
```

Comment:
1 Transfer DSMetadata queries to the metric profile.

2. Relocate Container last active date queries to the metric 
profile and update the code to reference queries using the metric profile.

3. Adjust requests and limit queries to avoid using query_range.